tests/TestCase/Database/Log/QueryLoggerTest.php

```diff
@@ -128,6 +128,26 @@ public function testStringInterpolationNamed()
     }
 
     /**
+     * Tests that placeholders are replaced with correctly escaped strings
+     *
+     * @return void
+     */
+    public function testStringInterpolationSpecialChars()
+    {
+        $logger = $this->getMockBuilder('\Cake\Database\Log\QueryLogger')
+            ->setMethods(['_log'])
+            ->getMock();
+        $query = new LoggedQuery;
+        $query->query = 'SELECT a FROM b where a = :p1 AND b = :p2 AND c = :p3 AND d = :p4';
+        $query->params = ['p1' => '$2y$10$dUAIj', 'p2' => '$0.23', 'p3' => 'a\\0b\\1c\\d', 'p4' => "a'b"];
+
+        $logger->expects($this->once())->method('_log')->with($query);
+        $logger->log($query);
+        $expected = "duration=0 rows=0 SELECT a FROM b where a = '\$2y\$10\$dUAIj' AND b = '\$0.23' AND c = 'a\\\\0b\\\\1c\\\\d' AND d = 'a''b'";
```

Comment:
SQL search for string `It's m\n\o` is:

1. `SELECT a FROM b WHERE c = 'It's m\n\o'` - current QuerryLogger emulated query
2. `SELECT a FROM b WHERE c = 'It''s m\\n\\o'` - valid in MySQL/PostgreSQL
3. `SELECT a FROM b WHERE c = 'It''s m\n\o'` - valid in Sqlite/SQL Server

My commit changed SQL to 2nd variant because I work with MySQL and I didn't thought about other DBs. But after I checked syntax of string literal in other DBs, now I'm thinking which variant is best.

QueryLogger is not DB specific. Emulated queries are not executed so security is low priority. But sometimes we need test queries externally if CakePHP works not as expected. I want it to be as much as possible evenly helpful for every DB engine developer. 

So I ask which variant is the best? Do CakePHP have higher priorities to some DB engine?